java-checks/src/main/java/org/sonar/java/checks/NullShouldNotBeUsedWithOptionalCheck.java

```diff
@@ -163,10 +164,21 @@ private static boolean isNull(ExpressionTree expression) {
     return expression.is(Tree.Kind.NULL_LITERAL);
   }
 
-  private void checkNullableAnnotation(ModifiersTree modifiers, String messageFormat) {
-    NullableAnnotationUtils.nullableAnnotation(modifiers)
-      .ifPresent(annotation -> context.reportIssue(this, annotation,
-        String.format(messageFormat, "@" + annotation.annotationType().symbolType().name())));
+  private void checkNullability(Symbol symbol, String messageFormat) {
+    SymbolMetadata.NullabilityLevel level;
+    if (symbol.isVariableSymbol()) {
+      level = SymbolMetadata.NullabilityLevel.VARIABLE;
+    } else {
+      level = SymbolMetadata.NullabilityLevel.METHOD;
+    }
+    SymbolMetadata.NullabilityData nullabilityData = symbol.metadata().nullabilityData();
+    if (nullabilityData.isNullable(level, false, false)) {
+      context.reportIssue(
+        this,
+        nullabilityData.declaration(),
+        String.format(messageFormat, "@" + nullabilityData.annotation().symbol().name())
```

Comment:
`nullabilityData.annotation()` is nullabe. Even if I don't think we can come up with an example to would make this code break, I think it makes sense to check it for defensive programming.
Or maybe use the helper method that I added in the [other PR](https://github.com/SonarSource/sonar-java/pull/3852/files#diff-b8e775a1b8262a3f713e6394e3f70eb7d5423ebf44500a19533cee068a1d2e0dR31)?